Photochemistry/src/ReactionNetwork.jl

```diff
@@ -8,87 +8,100 @@
 #                      Functions to load and manipulate the network             #
 #===============================================================================#
 
-function calculate_and_write_column_rates(rxn_filename, atm_state; globvars...)
+function calculate_and_write_column_rates(rxn_filename, atm_state, n_horiz; globvars...)
     #=
-    Calculates the column rates for all the reactions in rxn_filename on the species densities in atm_state 
-    and writes them back out in a column to rxn_filename. Loops over each sheet. 
+    Calculates the column rates for all reactions in rxn_filename based on species densities in atm_state
+    and writes them back out into rxn_filename. Creates separate columns for each horizontal position (ihoriz).
     =#
-    
+
     GV = values(globvars)
-    required = [:all_species, :dz, :ion_species, :num_layers, :reaction_network, :results_dir, :sim_folder_name, :Tn, :Ti, :Te]
+    required = [:all_species, :dz, :ion_species, :num_layers, :reaction_network,
+                :results_dir, :sim_folder_name, :Tn, :Ti, :Te]
     check_requirements(keys(GV), required)
-    
+
     flush(stdout)
     println("Writing out column rates to the reaction log...")
     flush(stdout)
-    
-    # Get the Jrates, needed later
-    Jratedict = Dict([j=>atm_state[j] for j in keys(atm_state) if occursin("J", string(j))]);
-    
-    Mtot = n_tot(atm_state; GV.all_species);
-    
-    # Open the active reactions file
-    the_spreadsheet_file = GV.results_dir*GV.sim_folder_name*"/"*rxn_filename
+
+    # Jrates needed for photochemical reactions
+    Jratedict = Dict([j => atm_state[j] for j in keys(atm_state) if occursin("J", string(j))])
+
+    # Loop through sheets in the spreadsheet
+    the_spreadsheet_file = GV.results_dir * GV.sim_folder_name * "/" * rxn_filename
     active_rxns_original = XLSX.readxlsx(the_spreadsheet_file)
     original_sheets = XLSX.sheetnames(active_rxns_original)
-    
-    # Loop through the sheets
-    for (j, sheet) in enumerate(original_sheets)
 
-        if occursin("Unused", sheet)
+    for sheet in original_sheets
+        if occursin("Unused", sheet) || occursin("Sheet1", sheet)
```

Comment:
I'm guessing Sheet1 here is a blank sheet - this is a bug where these get written out sometimes. This is fine for now but when everything is settled we should try to control the bug where it writes out a blank sheet first before the sheets with content.

Reply:
Oh wait, I see, I originally had the same logic but divided into two statements. Okay this change is good, never mind.